src/lightly_train/_task_models/dinov2_eomt_semantic_segmentation/task_model.py

```diff
@@ -415,13 +415,23 @@ def untile(
     ) -> list[Tensor]:
         logit_sums, logit_counts = [], []
 
+        dtype = crop_logits.dtype
+
         # Initialize the tensors containing the final predictions.
         for size in image_sizes:
             logit_sums.append(
-                torch.zeros((crop_logits.shape[1], *size), device=crop_logits.device)
+                torch.zeros(
+                    (crop_logits.shape[1], *size),
+                    device=crop_logits.device,
+                    dtype=dtype,
+                )
             )
```

Comment:
Nit (doesn't need to be addressed): The "modern" torch way to do this is:
```
crop_logits.new_zeros((crop_logits.shape[1], *size))
```
Then the new tensor is automatically on the correct device with the correct dtype.